addons/project_timesheet_holidays/models/hr_holidays.py

```diff
@@ -65,32 +95,14 @@ def _validate_leave_request(self):
                 lambda request: request.holiday_type == 'employee' and
                                 request.holiday_status_id.timesheet_project_id and
                                 request.holiday_status_id.timesheet_task_id):
-            holiday_project = holiday.holiday_status_id.timesheet_project_id
-            holiday_task = holiday.holiday_status_id.timesheet_task_id
+            holiday._create_timesheet_lines()
 
-            work_hours_data = holiday.employee_id.list_work_time_per_day(
-                fields.Datetime.from_string(holiday.date_from),
-                fields.Datetime.from_string(holiday.date_to),
-            )
-            for index, (day_date, work_hours_count) in enumerate(work_hours_data):
-                self.env['account.analytic.line'].create({
-                    'name': "%s (%s/%s)" % (holiday.name or '', index + 1, len(work_hours_data)),
-                    'project_id': holiday_project.id,
-                    'task_id': holiday_task.id,
-                    'account_id': holiday_project.analytic_account_id.id,
-                    'unit_amount': work_hours_count,
-                    'user_id': holiday.employee_id.user_id.id,
-                    'date': fields.Date.to_string(day_date),
-                    'holiday_id': holiday.id,
-                    'employee_id': holiday.employee_id.id,
-                })
-
-        return super(Holidays, self)._validate_leave_request()
+        return super()._validate_leave_request()
 
     @api.multi
     def action_refuse(self):
         """ Remove the timesheets linked to the refused holidays """
-        result = super(Holidays, self).action_refuse()
+        result = super().action_refuse()
```

Comment:
I think this kind of change shouldn't be in the PR... Same for the other super above.

Reply:
On other hand, why to keep useless code in codebase? Or wait for Odoo employees to do that?

Reply:
Yeah, they do that on future refactorings and cleaning ups...